src/components/manager.ts

```diff
@@ -200,155 +227,139 @@ export class Manager {
         const rootFilesExcludePatterns = configuration.get('latex.search.rootFiles.exclude') as string[]
         const rootFilesExcludeGlob = rootFilesExcludePatterns.length > 0 ? '{' + rootFilesExcludePatterns.join(',') + '}' : undefined
         try {
-            const urls = await vscode.workspace.findFiles(rootFilesIncludeGlob, rootFilesExcludeGlob)
-            for (const url of urls) {
-                const content = utils.stripComments(fs.readFileSync(url.fsPath).toString(), '%')
+            const files = await vscode.workspace.findFiles(rootFilesIncludeGlob, rootFilesExcludeGlob)
+            for (const file of files) {
+                const content = utils.stripComments(fs.readFileSync(file.fsPath).toString(), '%')
```

Comment:
This change does not solve the problem since the default encoding of `buf.toString` is utf-8. See [link](https://nodejs.org/api/buffer.html#buffer_buf_tostring_encoding_start_end).

Reply:
`stripComments` is implemented by @jlelong several months ago. Need to wait for him to answer this.

Reply:
Additionally, we intensively use `fs.readFileSync(someFile).toString()` throughout the extension. I am not aware of other approaches to read a text file into string. All in all we need to guess an encoding, or find a way (I have no idea) to determine one.

Reply:
I do not know why this code does not cause problems.  As long as no issue is reported by users, I think it is OK.

However, if you expect the newly introduced API, `getContent`, will be used frequently in LaTeX Workshop, it would cause problems.

Reply:
The `stripComments` function simply remove everything from a comment sign to the end of line by taking care of escaped comment signs. My 2 cents on the encoding issue
- we mostly only search strings of the form `\ + ascii_char` and this is not sensitive to the encoding
- `utf-8` has become widely used
- We may troubles with labels and file names (but I am not sure `latex` handles non ascii file names well).

Is `getContent` not restricted to getting the content or an editor or can it read a file from the disk?